src/backend/parser/analyze.c

```diff
@@ -3506,6 +3510,44 @@ applyLockingClause(Query *qry, Index rtindex,
 	qry->rowMarks = lappend(qry->rowMarks, rc);
 }
 
+/*
+ * Attach column names from a ColumnDef list to a TargetEntry list
```

Comment:
The function seems to assume that items in `ColumnDef` list and `TargetEntry` list are in the same order according to their attribute numbers.  Moreover, `TargetEntry` list may contain out-of-order junk attributes.  Is that assumption solid enough?  Is it possible to add assertions such as `(d->resorigcol == s->attnum)` to state the assumption more explicitly?

Reply:
This function is from PG 9.1 and has been removed from PG 9.2 and it is mis-used at this place. At least, `into->colNames` is a list of `T_String` rather than `ColumnDef`.
I refreshed the PR and removed this function. Thanks for reviewing.